src/MockNet/Http/Internal/HttpMessageHandler.cs

```diff
@@ -20,17 +20,19 @@ protected override async Task<SystemHttpResponseMessage> SendAsync(SystemHttpReq
         {
             var setups = this.setups.Find(request).ToList();
 
-            if (!setups.Any())
+            // there are no setups that were found successful.
+            if (!setups.Any(x => x.Exception is null))
             {
-                throw await MockHttpClientException.NoMatchingRequestsAsync(request);
+                throw MockHttpClientException.Combined(setups.Select(x => x.Exception));
             }
 
-            if (setups.Count() > 1)
+            // there are more than 1 setup that was found.
+            if (setups.Count(x => x.Exception is null) > 1)
```

Comment:
This and the previous `if`, would it be the same if we do:

```csharp
if (setups.Any(x => x.Exception is null))
{
  throw await MockHttpClientException.MatchedMultipleRequests(request, setups.Count());
}
else
{
  throw MockHttpClientException.Combined(setups.Select(x => x.Exception));
}
```

Reply:
`setups.Any(x => x.Exception is null)` means if there was at least 1 successful setup, do not throw an exception, but if there are more than 1 `setups.Count(x => x.Exception is null) > 1` successful setups throw an exception.  

I don't know if you proposed the `if` statement would work because this would always throw either a combined exception or the matched multiple request exception.

Reply:
🤔 the only solution I can think of is to add a test to check for exactly this case; in fact, I'd suggest adding that test regardless of which code we use, just asking ourselves this question makes me think we're missing it.

Again, regardless of which solution we use, I think we need to add more comprehensive comments to this method.

Reply:
Added: 
`Theorem.MockNet.Http.Tests.MockHttpClientExceptionTests.MockHttpClientExceptionIsThrownIfMultipleUnmatchesOccur` for multiple setups that all fail

`Theorem.MockNet.Http.Tests.Test.TestMultipleMocksWithOneSuccessful` for mulitple setups with one successful